pkgs/top-level/all-packages.nix

```
@@ -9404,6 +9404,15 @@ with pkgs;
 
   nodejs-slim = nodejs-slim_18;
 
+  # The NodeJS packages has to be build in 32bit environment if host platform is
+  # also 32bit because it uses 32bit stubs and links against 32bit OpenSSL. The
+  # only architecture that supports execution of 32bit is x86_64 and thus that
```

```suggestion
  # only architecture that generally supports execution of 32bit is x86_64 and thus that
```

IIRC AArch64 can execute AArch32 code if enabled in the kernel configuration ([Kernel support for 32-bit EL0 found in arch/arm64/Kconfig](https://cateee.net/lkddb/web-lkddb/COMPAT.html))? That seems to be CPU-dependent for ARM, though I think 32-bit can also be disabled on x86-64.

---

Yes, that is CPU-dependent on aarch64 because it relies on the CPU to actually interpret those arm instructions, as far as I know. The only way I know is to use binfmt and register an emulator. On the second note about disable of x86: you can even theoretically disable the https://www.kernelconfig.io/CONFIG_COMPAT_BINFMT_ELF?kernelversion=6.3.12 it just depends on where you draw the line between "this is expected platform support and this is not". Linux kernel provides way too many compilations knobs to cover all corners.

But I agree with the comment change (and it is done by now). This answer is only just a polemic.